### PR TITLE
Trigger mouse over on canvas enter

### PR DESCRIPTION
**Pull request by stefanhayden**
As talked about in #3384 this adds the existing `mouse:over` event to fire when the mouse enters the canvas.

If the mouse enters the canvas and there is no target the fire the event. We need to do a check if there is a target, other wise the event will fire 2 times on entering the canvas with an object on the edge. once with null and once with the target,

### CODE REVIEW

File: src/mixins/canvas_events.mixin.js [hunk 1]
```diff
@@ -141,6 +144,19 @@
...
     /**
      * @private
+     * @param {Event} e Event object fired on mouseenter
+     */
+    _onMouseEnter: function(e) {
+      if (!this.findTarget(e)) {
+        var target = this._hoveredTarget || null;
+        this.fire('mouse:over', { target: target, e: e });
+        this._hoveredTarget = null;
+        target && target.fire('mouseenter', { e: e });
```
Comment (asturur) [inline src/mixins/canvas_events.mixin.js]: i think ```mouseenter``` is our ```mouseover```.

But beside that, if you do not find any target, ```this.findTarget(e)``` why do you want to fire the event for the last hoverdTarget ?

**Comment by asturur** [comment]
Thanks @stefanhayden i left a comment on the code, i m not sure of something.

**Comment by stefanhayden** [comment]
you were right. So we only fire this event if there is no target. we also set the _hoveredTarget  to null or mouse:out reports undefined instead of null for the target.

**Comment by asturur** [comment]
nice thanks!

**Pull request merged**